src/trio/_highlevel_socket.py

```diff
@@ -352,19 +358,34 @@ class SocketListener(Listener[SocketStream]):
     incoming connections as :class:`SocketStream` objects.
 
     Args:
+
       socket: The Trio socket object to wrap. Must have type ``SOCK_STREAM``,
           and be listening.
 
+      path: Used for keeping track of which path a Unix socket is bound
+          to. If not ``None``, :meth:`aclose` will unlink this path.
+          File must have socket mode flag set.
+
     Note that the :class:`SocketListener` "takes ownership" of the given
     socket; closing the :class:`SocketListener` will also close the socket.
 
     .. attribute:: socket
 
        The Trio socket object that this stream wraps.
 
+    .. attribute:: path
+
+       The path to unlink in :meth:`aclose` that a Unix socket is bound to.
+
     """
 
-    def __init__(self, socket: SocketType) -> None:
+    __slots__ = ("path", "socket")
+
+    def __init__(
+        self,
+        socket: SocketType,
+        path: str | bytes | PathLike[str] | PathLike[bytes] | None = None,
```

Comment:
I think I preferred how the old version didn't require this `path` argument. Is this really required? I assume we can just check whether a socket is unix type with `socket.family == getattr(tsocket, "AF_UNIX", None)` and have the old logic to get the path.

Reply:
Was looking deeper into things noted in #279 and saw this comment: https://github.com/python-trio/trio/issues/279#issuecomment-322114125

Reply:
Ah, but I missed something, we only have to care about that if we are renaming socket files, which this implementation does not do, so it would be fine to do the original thing.